src/container/getRelayQueries.js

```diff
@@ -36,14 +37,19 @@ function getRelayQueries(
   Component: RelayLazyContainer,
   route: RelayQueryConfigInterface
 ): RelayQuerySet {
-  let cache = queryCache.get(Component);
-  if (!cache) {
-    cache = {};
-    queryCache.set(Component, cache);
-  }
-  const cacheKey = route.name + ':' + stableStringify(route.params);
-  if (cache.hasOwnProperty(cacheKey)) {
-    return cache[cacheKey];
+  let cache = {};
+  let cacheKey = '';
+  const queryCacheEnabled = DisableRelayQueryCache.getCacheEnabled();
+  if (queryCacheEnabled) {
+    cache = queryCache.get(Component);
+    if (!cache) {
+      cache = {};
+      queryCache.set(Component, cache);
+    }
+    cacheKey = route.name + ':' + stableStringify(route.params);
+    if (cache.hasOwnProperty(cacheKey)) {
+      return cache[cacheKey];
+    }
```

Comment:
Let's break this into two methods: one that computes the query set (something like `buildQuerySet()`) and then a method (`getRelayQueries`) that calls it and knows about caching.